### ISSUE TITLE
Can't load cache in memory without syncing / can't access data offline

### ISSUE THREAD

**Issue by rstoye** [wmeints/SyncFrameworkToolkit#3]
Hi there,

we successfully implemented this SyncFramework into our test app. We can sync both ways without any problems. However, if we want to access the data on the client without syncing, the tablecollections don't contain any data.

So for example we get a new phone, we sync for the first time and get 200 changes from the server. Those 200 changes are stored in /data/data/DemoApp.DemoApp/files/.config/.isolated-storage/TestDb/000001.D in JSON format.

Now we restart our phone for whatever reason, we start our app and now want to access the data that's stored on the phone without syncing. From what I understand that's what the method "Load()" from the "IsolatedStorageOfflineContext" is for. And by reading through the code you can tell that this method at some point loads the 000001.D file, parses the json and...well...technically should fill up the tablecollections.

To some extend it does exactly that. If I count all the items in my collection I get the 200 results as expected, however if I open each individual item all the properties inside are NULL.

Am I missing something here? Is syncing upon starting the app always required? What if I don't have an internet connection? Or is this simply a bug?

Any help would be appreciated.

**Comment by wmeints** [comment]
Normally you don't to access the offline providers directly. So that might be the reason why some pieces of data or not loaded when you go down that route.

**Comment by rstoye** [comment]
Hello there,

yeah that's the conclusion we came to after several days spent studying the source code of your framework. We also found out that the local json file on the phone only contains primary key data of the database, which explains why we couldn't load anything into the collections without synching first. I'm not sure if it's supposed to be like that or if we made a mistake at some point.

Anyway we switched over to Siaqodb (a NoSQL embedded object and document database engine), which is not free, but also supports synchronizing data via the Microsoft Sync Framework in a Xamarin Android Project and we managed to get it to work, so we're now using that.

Still, thanks for responding.
